SKYKitChat/Classes/UI/ConversationView/SKYChatConversationViewController.swift

```diff
@@ -291,7 +296,52 @@ extension SKYChatConversationViewController {
         self.messages.append(msg)
         self.finishSendingMessage(animated: true)
 
-        // TODO: trigger finish typing event
+    }
+}
+
+
+// MARK: - Subscription
+
+extension SKYChatConversationViewController {
+
+    open func subscribeMessageChanges() {
+        if self.messageChangeHandler == nil {
```

Comment:
When I wrote `subscribeMessageChanges()` and `subscribeTypingIndicatorChanges()`, I am thinking this will be called multiple times and the handlers can be reused or even updated by subclasses. 

But finally, `subscribeMessageChanges()` and `subscribeTypingIndicatorChanges()` are not able to be called multiple times. I will make some updates to achieve that.